### PR TITLE
Yro/sandbox update

### PR DESCRIPTION
**Pull request by yro**
### Description
Update/Bugfix for sandbox

### Testing
- [ ] Unit tests

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @muzaffaryousaf 

FYI: @edx/educator-neem

### Post-review
- [ ] Rebase and squash commits

### CODE REVIEW

File: control/veda_deliver.py [hunk 1]
```diff
@@ -410,15 +408,11 @@ def _BOTO_SINGLEPART(self):
...
-                self.auth_dict['edx_access_key_id'],
-                self.auth_dict['edx_secret_access_key']
-            )
+            conn = boto.connect_s3()
             delv_bucket = conn.get_bucket(
                 self.auth_dict['edx_s3_endpoint_bucket']
             )
-
-        except:
+        except S3ResponseError:
```
Comment (muzaffaryousaf) [inline control/veda_deliver.py]: Do we know what other exceptions could be raised ?

File: control/veda_deliver.py [hunk 2]
```diff
@@ -472,12 +466,9 @@ def _BOTO_MULTIPART(self):
         Connect to s3
         """
         try:
-            c = boto.connect_s3(
-                settings['aws_access_key'],
-                settings['aws_secret_key']
-            )
+            c = boto.connect_s3()
             b = c.lookup(settings['aws_deliver_bucket'])
```
Comment (muzaffaryousaf) [inline control/veda_deliver.py]: settings are not accessible here.

File: control/veda_file_discovery.py [hunk 3]
```diff
@@ -62,25 +63,20 @@ def __init__(self, **kwargs):
...
-        """
-        Crawl ingest bucket looking for files
-        """
-        conn = S3Connection(
-            self.auth_dict['veda_access_key_id'],
-            self.auth_dict['veda_secret_access_key']
-        )
+            return
+
+        conn = S3Connection()
```
Comment (muzaffaryousaf) [inline control/veda_file_discovery.py]: can we use `boto.connect_s3()` for consistency ?

File: control/veda_file_discovery.py [hunk 4]
```diff
@@ -146,36 +139,30 @@ def about_video_validate(self, meta, key):
...
-        """
-        Ingest files from studio upload endpoint
-        """
-        conn = S3Connection(
-            self.auth_dict['edx_access_key_id'],
-            self.auth_dict['edx_secret_access_key']
-        )
-
-        """Occassional s3 Error"""
+        conn = S3Connection()
```
Comment (muzaffaryousaf) [inline control/veda_file_discovery.py]: Same as above for consistency.

File: control/veda_file_discovery.py [hunk 5]
```diff
@@ -225,7 +212,7 @@ def studio_s3_validate(self, meta, key):
                     )
                 )
                 file_ingested = True
-            except:
+            except S3ResponseError:
```
Comment (muzaffaryousaf) [inline control/veda_file_discovery.py]: Same exception will be raised by calling  `meta.get_contents_to_filename` ?

File: control/veda_file_discovery.py [hunk 6]
```diff
@@ -237,7 +224,7 @@ def studio_s3_validate(self, meta, key):
...
```
Comment (muzaffaryousaf) [inline control/veda_file_discovery.py]: Same as above.

File: control/veda_file_discovery.py [hunk 7]
```diff
@@ -270,17 +257,18 @@ def studio_s3_validate(self, meta, key):
...
-            key.copy(self.bucket, new_key)
-        except:
-            key.copy(self.bucket, new_key)
-        # key.copy(self.bucket, new_key)
+        if self.auth_dict['edx_s3_ingest_prefix'] is not None:
+            new_key = 'prod-edx/processed/' + key.name[::-1].split('/')[0][::-1]
+            try:
+                key.copy(self.bucket, new_key)
+            except S3ResponseError:
+                key.copy(self.bucket, new_key)
```
Comment (muzaffaryousaf) [inline control/veda_file_discovery.py]: I think this will not work either. Will it ?

File: control/veda_hotstore.py [hunk 8]
```diff
@@ -1,13 +1,16 @@
...
-import yaml
-import shutil
-from os.path import expanduser
+from boto.exception import S3ResponseError
 import newrelic.agent
+import os
+from os.path import expanduser
+import shutil
+import sys
+import yaml
```
Comment (muzaffaryousaf) [inline control/veda_hotstore.py]: Pep8 prefers the library imports before the application's imports. Also, we're adding isort to sort these imports which also make sure that imports are alphabetically ordered.

File: VEDA_OS01/fixtures/destinations.json [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+      "tp_proc": false,
+      "c24_proc": false,
+      "s3_proc": true,
+      "xue": true,
+      "local_storedir": "course-v1:VEDA+VEDA201+2015_T1",
+      "xuetang_proc": false,
+      "studio_hex": "xxxx"
+    }
+  }
+]
```
Comment (yro) [inline VEDA_OS01/fixtures/destinations.json]: @muzaffaryousaf It also adds the encode profiles, which are usually unchanged.

File: control/veda_deliver.py [hunk 10]
```diff
@@ -265,10 +266,7 @@ def _CLEANUP(self):
                 self.auth_dict['edx_s3_endpoint_bucket']:
             return None
 
-        conn = boto.connect_s3(
-            self.auth_dict['veda_access_key_id'],
-            self.auth_dict['veda_secret_access_key']
-        )
```
Comment (yro) [inline control/veda_deliver.py]: @muzaffaryousaf Correct. It's done in edx-video-worker, but not here.

File: control/veda_deliver.py [hunk 11]
```diff
@@ -265,10 +266,7 @@ def _CLEANUP(self):
...
+        conn = boto.connect_s3()
```
Comment (yro) [inline control/veda_deliver.py]: @muzaffaryousaf we probably should.

File: control/veda_deliver.py (hunk 1 above)
Comment (yro) [inline control/veda_deliver.py]: Refactoring...

File: control/veda_deliver.py (hunk 2 above)
Comment (yro) [inline control/veda_deliver.py]: Good catch, refactoring.

File: control/veda_file_discovery.py (hunk 3 above)
Comment (yro) [inline control/veda_file_discovery.py]: 👍

File: control/veda_file_discovery.py (hunk 7 above)
Comment (yro) [inline control/veda_file_discovery.py]: No, there's a typo in the indentation too.

Comment (yro) [inline control/veda_file_discovery.py]: @muzaffaryousaf Changed.

File: control/veda_hotstore.py (hunk 8 above)
Comment (yro) [inline control/veda_hotstore.py]: 👍

**Comment by yro** [comment]
@muzaffaryousaf Is the expectation for these tests to pass as well?

**Comment by muzaffaryousaf** [comment]
@yro No, we've not got the chance to write tests for this repo yet but maybe in future when we've bandwidth.

**Comment by yro** [comment]
@muzaffaryousaf Thanks! No worries, just making sure I wasn't going to jump the gun.

**Pull request merged**